src/pip/_internal/req/constructors.py

```diff
@@ -236,26 +289,9 @@ def install_req_from_line(
         link = Link(name)
     else:
         p, extras_as_string = _strip_extras(path)
-        looks_like_dir = os.path.isdir(p) and (
-            os.path.sep in name or
-            (os.path.altsep is not None and os.path.altsep in name) or
-            name.startswith('.')
-        )
-        if looks_like_dir:
-            if not is_installable_dir(p):
-                raise InstallationError(
-                    "Directory %r is not installable. Neither 'setup.py' "
-                    "nor 'pyproject.toml' found." % name
-                )
-            link = Link(path_to_url(p))
-        elif is_archive_file(p):
-            if not os.path.isfile(p):
-                logger.warning(
-                    'Requirement %r looks like a filename, but the '
-                    'file does not exist',
-                    name
-                )
-            link = Link(path_to_url(p))
+        url = _get_path_to_url(p, name)
```

Comment:
@vinicyusmacedo Another review comment I made previously is that `_get_url_from_path()` would be a better name than `_get_path_to_url ()` for this because the function returns a URL (or None), given a path. `_get_path_to_url()` doesn't have a clear meaning to it.